tests/app/main/views/test_suppliers.py

```diff
@@ -202,6 +202,57 @@ def test_if_most_recent_framework_has_no_declaration_show_account_company_detail
             }
         }
 
+    @mock.patch("app.main.views.suppliers.render_template", return_value="")
+    def test_if_most_recent_framework_has_old_declaration_show_account_company_details(self, render_template):
+        self.data_api_client.get_supplier_frameworks.return_value = {
+            "frameworkInterest": [
+                SupplierFrameworkStub(
+                    framework_slug="g-cloud-8",
+                    declaration={
+                        "dunsNumber": mock.sentinel.duns_number,
+                        "currentRegisteredCountry": "country:FR",
+                        "registeredVATNumber": "12345678",
+                        "registeredAddressBuilding": "123 Rue Morgue",
+                        "registeredAddressTown": "Paris",
+                        "registeredAddressPostcode": "76876",
```

Comment:
I think the problem we were having was slightly different, in that there was a declaration, but it didn't have any of the fields above, just the application answers. However the supplier in question has since been updated, and I can't find another example! Perhaps this was a one off?

Reply:
Interesting...

I guess this fix should still work? I can add another test scenario

Reply:
OK. It's G7 where the company data isn't included in the declaration (but the declaration has other stuff in it).

The G7 declarations are pretty different to the subsequent ones so we shouldn't use them for fetching company details. They'll look something like:

```
"declaration": {
        "AQA3": true,
        "PR1": true,
        "PR2": true,
        "PR3": true,
        "PR4": true,
        "PR5": true,
        ...
}
```

Admins still want to view them, but we won't be able to get the DUNS etc from there.